src/modules/user/controller/index.js

```diff
@@ -266,8 +267,91 @@ async function verifyUser(req, res) {
   });
 }
 
+/**
+ *
+ * @param {*} req ExpressRequest
+ * @param {*} res ExpressResponse
+ * @returns {*} token and useremail || validation errors
+ */
+// a function to reset password
+const initateResetPassword = (req, res) => {
+  const { email } = req.body;
+  const Transport = nodemailer.createTransport({
+    service: process.env.MAIL_HOST,
+    auth: {
+      user: process.env.MAIL,
+      pass: process.env.MAIL_PASSWORD
+    }
+  });
+  const mailOptions = {
+    to: email,
+    subject: 'Barefoot Nomad Reset password',
+    html: '<p>Welcome to barefoot Nomad, Click the link below to reset password.</p><a href= \'http://localhost:5000/api/v1/users/reset-password\'><b>Click to reset</b> </a>',
+  };
+
+  Transport.sendMail(mailOptions, (error) => {
+    if (error) {
+      console.log(error);
+    } else {
+      console.log('Email sent successfully');
+    }
+    return res.status(200).json({
+      statusCode: 'OK',
+      message: 'Email sent successfully'
+
+    });
+  });
+};
+
+/**
+ *
+ * @param {*} req ExpressRequest
+ * @param {*} res ExpressResponse
+ * @returns {*} success message or error message
+ */
+async function resetPassword(req, res) {
+  const { email, currentPassword, newPassword } = req.body;
+
+  const user = await User.findOne({
+    where: {
+      email
+    }
+  });
+
+  if (!user) {
+    return res.status(400).json({
+      statusCode: 'BAD_REQUEST',
+      errors: {
+        email: [
+          'User not found'
```

Comment:
As this feature is related to security, you can create a uniform error message that someone (hackers) can't figure out to know what is wrong.